src/tools/quemap/lightmap.c

```diff
@@ -620,6 +620,87 @@ static _Bool NudgeSamplePosition(const vec3_t in, const vec3_t normal, const vec
 	return Light_PointPVS(out, pvs);
 }
 
+/**
+ * @brief Trace a ray and gather indirect light from surfaces
+ */
+static void GatherSampleBounceTrace(int8_t bounce, const vec3_t center, const vec3_t pos, const vec3_t normal, const vec_t dist, vec_t *sample) {
+
+	vec3_t end;
+	vec_t dot, sqdist;
+	vec3_t color;
+	vec_t hardness;
+	vec3_t bounce_dir, bounce_pos;
+	vec_t bounce_dist;
+	cm_trace_t trace;
+
+	// get a trace
+
+	VectorScale(normal, dist, end);
+
+	Light_Trace(&trace, pos, end, CONTENTS_SOLID);
+
+	if (!trace.surface) {
+		return;
+	}
+
+	if (trace.start_solid) {
+		return;
+	}
+
+	// distance squared and trace hit normal dot product
+
+	sqdist = trace.fraction * trace.fraction;
+	dot = DotProduct(trace.plane.normal, normal);
+
+	// get surface color
+
+	const cm_material_t *mat = trace.surface->material;
+	if (mat) {
+		hardness = Clamp(mat->hardness, 0.0, 5.0);
+		GetTextureColor(mat->diffuse, color);
+	} else {
+		hardness = 1.0;
+		VectorSet(color, 1.0, 1.0, 1.0);
+	}
+
+	const vec_t reflect = hardness / 5.0;
+	VectorScale(color, reflect * sqdist, color);
+
+	if ((trace.fraction * dist) < AMBIENT_OCCLUSION_DIST) {
+		vec_t ao = (1.0 - reflect) * (trace.fraction * (dist / AMBIENT_OCCLUSION_DIST)) * Max(0.001, dot);
+		color[0] -= ao;
+	}
+
+	// multiplicative color
+
+	sample[0] *= color[0];
+	sample[1] *= color[1];
+	sample[2] *= color[2];
```

Comment:
I'm using this method mainly because for each sample (Think -light -extra = 5 samples per lxuel) has X traces (8 right now I think), they have to be divided down so they only affect the final result by the same amount as 1 sample and 1 trace, ints might not work too well here.